include/odp/api/spec/schedule_types.h

```diff
@@ -159,6 +167,87 @@ typedef	struct odp_schedule_param_t {
 	uint32_t lock_count;
 } odp_schedule_param_t;
 
+/**
+ * Scheduler capabilities
+ */
+typedef struct odp_schedule_capability_t {
+	/** Maximum number of ordered locks per queue */
+	uint32_t max_ordered_locks;
+
+	/** Maximum number of scheduling groups */
+	uint32_t max_groups;
+
+	/** Number of scheduling priorities */
+	uint32_t max_prios;
+
+	/** Maximum number of scheduled (ODP_BLOCKING) queues of the default
+	 * size. */
+	uint32_t max_queues;
+
+	/** Maximum number of events a scheduled (ODP_BLOCKING) queue can store
+	 * simultaneously. The value of zero means that scheduled queues do not
+	 * have a size limit, but a single queue can store all available
+	 * events. */
+	uint32_t max_queue_size;
+
+	/** Maximum supported flows per queue.
+	 * Specifies the maximum number of flows per queue supported by the
+	 * implementation.  A value of 0 indicates flow aware mode is not
+	 * supported.
+	 */
+	uint32_t max_flows;
+
+	/** Lock-free (ODP_NONBLOCKING_LF) queues support.
+	 * The specification is the same as for the blocking implementation. */
+	odp_support_t lockfree_queues;
+
+	/** Wait-free (ODP_NONBLOCKING_WF) queues support.
+	 * The specification is the same as for the blocking implementation. */
+	odp_support_t waitfree_queues;
+
+} odp_schedule_capability_t;
+
+/**
+ * Schedule configuration
+ */
+typedef struct odp_schedule_config_t {
+	/** Maximum number of scheduled queues to be supported.
+	 *
+	 * @see odp_schedule_capability_t
+	 */
+	uint32_t num_queues;
+
+	/** Scheduled queue config */
+	struct {
+		/** Maximum number of events required to be stored
+		 * simultaneously in scheduled queue. This number must not
+		 * exceed 'capa.queue.max_size' required by the application.  A
+		 * value of 0 configures default queue size supported by the
+		 * implementation.
+		 */
+		uint32_t queue_size;
+
+		/** Number of flows per queue to be supported.  Scheduler
+		 * enables flow aware mode when flow count is configured
+		 * greater than 1 (up to 'capa.queue.max_flows').
+		 *
+		 * Flows are lightweight entities and events can be assigned to
+		 * specific flows by the application using
+		 * odp_event_flow_id_set() before enqueuing the event into the
+		 * scheduler. This value is ignored unless scheduler supports
+		 * flow aware mode.
+		 *
+		 * This number should be less than maximum flow supported by
+		 * the implementation. The default value is zero.
+		 *
+		 * @see odp_schedule_capability_t
+		 */
+		uint32_t num_flows;
+
+	} queue;
```

Comment:
Now when capa is flat, config should be also. So, "struct queue" needs to be removed. Just all three values on the same (base level). Later we'll add num_prios, num_groups, etc also into this config_t.

Reply:
ok